SCClassLibrary/Common/Control/Buffer.sc

```diff
@@ -381,7 +386,7 @@ Buffer {
 
 	zero { arg completionMessage;
 		if(bufnum.isNil) { Error("Cannot call % on a % that has been freed".format(thisMethod.name, this.class.name)).throw };
-		server.listSendMsg([\b_zero, bufnum, completionMessage.value(this)])
+		server.listSendMsg(this.zeroMsg(completionMessage));
```

Comment:
many of these functions now needlessly check `bufnum.isNil` twice -- once in `xyz` and again in `xyzMsg`; can you please fix that?

Reply:
We added all those checks a while back. The reason for the duplication is that users can directly call the `xyzMsg` method when manually sending server messages. We wanted to always check for nil buffers and unfortunately that meant checking in both the `xyz` and `xyzMsg` methods. You're right that all these repeated if statements look pretty silly though. We should do something about that :)

Reply:
oh thanks, i did not think of that!

Reply:
I think the solution is to keep the check in the Msg methods, and remove it from the message-sending methods, and change the error text to be verb-centric rather than method-selector-centric -- "cannot write a buffer that has been freed" instead of "cannot call 'write' on a buffer that has been freed." The text still makes sense to the user and there's no need to differentiate the methods. So I will un-resolve this bit and maybe deal with it today.

Reply:
Thanks for taking the time to improve this!

Reply:
Done.